tests/legacy-cli/e2e_runner.ts

```diff
@@ -123,11 +123,19 @@ const tests = allTests.filter((name) => {
 const testsToRun = tests.filter((name, i) => shardId === null || i % nbShards == shardId);
 
 if (testsToRun.length === 0) {
-  console.log(`No tests would be ran, aborting.`);
-  process.exit(1);
+  if (shardId !== null && tests.length >= shardId ? 1 : 0) {
```

Comment:
I guess this is actually a fix... if there are more shards then tests it shouldn't fail.